### PR TITLE
Adding EnableFIPS flag in ManageMachinePool struct

### LINKED ISSUE #3752: Expose EnableFIPS flag for AKS clusters
/kind feature
Extending from https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/1701 for managed pool to have the following option.

**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]
// EnableFIPS - Whether to use FIPS enabled OS
// +optional
EnableFIPS

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- cluster-api-provider-azure version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by mjnovice**
**What type of PR is this?**
Aimed at fixing #3752 

/kind feature
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #3752 

**Special notes for your reviewer**:

- [ ] cherry-pick candidate

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
Adding EnableFIPS flag in ManageMachinePool struct
```

### CODE REVIEW

File: api/v1beta1/azuremanagedmachinepool_types.go [hunk 1]
```diff
@@ -556,6 +556,11 @@ type AzureManagedMachinePoolSpec struct {
 	// Immutable.
 	// +optional
 	SubnetName *string `json:"subnetName,omitempty"`
+
+	// EnableFIPS indicates whether FIPS is enabled on the node pool.
+	// Immutable.
```
Comment (Jont828) [inline api/v1beta1/azuremanagedmachinepool_types.go:561]: In order to make this immutable, you're going to have to add a block similar to [this](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/b07d34e72a9ff577c77342b7eb5d6c610c71335c/api/v1beta1/azuremanagedmachinepool_webhook.go#L170-L175) in `azuremanagedmachinepool_webhook.go`.

**Review by Jont828 — COMMENTED** [review]
Nice work, this is looking good so far! Is this PR ready or are you still looking to add some more?

**Comment by mjnovice** [comment]
> Nice work, this is looking good so far! Is this PR ready or are you still looking to add some more?

@Jont828 thanks for the review. I think i just want to enableFips. So don't want to add anything else in this

File: api/v1beta1/azuremanagedmachinepool_types.go (hunk 1 above)
Comment (mjnovice) [reply api/v1beta1/azuremanagedmachinepool_types.go:561]: Thanks for pointing that out @Jont828 . Added.

File: api/v1beta1/azuremanagedmachinepool_webhook.go [hunk 2]
```diff
@@ -174,6 +174,13 @@ func (mw *azureManagedMachinePoolWebhook) ValidateUpdate(ctx context.Context, ol
 		allErrs = append(allErrs, err)
 	}
 
+	if err := webhookutils.ValidateImmutable(
```
Comment (Jont828) [inline api/v1beta1/azuremanagedmachinepool_webhook.go:177]: Could you add a test for this block in [this file](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/1ab9b7448594f5466a1c74b95696cf9ce88adea3/api/v1beta1/azuremanagedmachinepool_webhook_test.go#L546-L559) as well?

Comment (mjnovice) [reply api/v1beta1/azuremanagedmachinepool_webhook.go:177]: done.

**Comment by Jont828** [comment]
Let's wait for the tests to finish running

**Comment by dtzar** [comment]
Looks like this is ready to merge now?

**Comment by mjnovice** [comment]
> Looks like this is ready to merge now?
@dtzar yeah. Not sure about how can I add release notes though.

**Comment by Jont828** [comment]
@mjnovice A couple more things! Please add a release note (you can edit the top PR comment and there's a section at the bottom) and also squash your commits!

**Comment by Jont828** [comment]
@CecileRobertMichon I think we're good to merge once the above tasks are done

**Comment by CecileRobertMichon** [comment]
@mjnovice could you please rebase your branch and make sure only your commit is showing up in commit history? Right now it has several commits that already merged in main which is also causing merge conflicts. I also recommend using a feature branch next time instead of your fork main branch (see https://www.kubernetes.dev/docs/guide/github-workflow/#3-create-a-working-branch)

**Comment by mjnovice** [comment]
@CecileRobertMichon sure. Have rebased the branch, will keep that in mind for future. 🙏

**Comment by mjnovice** [comment]
@Jont828 might need your approval for some of the workflows post rebase 🙏

**Comment by Jont828** [comment]
It looks like the jobs are triggered, we can give it a bit to see what the results are.

**Review by CecileRobertMichon — COMMENTED** [review]
/approve

**Pull request merged**